CONTRIBUTING.md

```diff
@@ -15,11 +15,36 @@ to discuss proposals and use pull requests to suggest changes. For more
 information see the [Cilium Development
 Guide](https://docs.cilium.io/en/latest/contributing/development/).
 
+## An important note about mutating cluster resources
+
+Cilium CLI must follow these general rules regarding mutating cluster resources:
+
+- Cilium CLI must not mutate resources that are not managed by the Cilium Helm
+  chart. Cilium CLI must use Cilium Helm values to configure Cilium installations.
+- The only exception to the rule above is the connectivity test command, which
+  may mutate resources within Kubernates namespaces with the name prefix specified
+  by --test-namespace flag. The connectivity test must not mutate resources that
+  affect network connectivity outside the test namespaces.
```

Comment:
I agree with these as a baseline. I think that these two principles can guard user clusters against dangerous changes. This is important for user trust - we can ask users to just run the CLI tests and the tool will provide safe feedback about the state of the cluster.

These guidelines can help with us thinking about how we build test functionality into the CLI in ways that are useful for both developers and users, rather than just solving problems that developers have during the dev cycle. Ideally we can get this CLI to tell us if the environment is set up correctly to properly handle a particular feature without mutating the cluster in any significant way. Smoke test of sorts.

I also recognize that these are quite restrictive and given that we already have the `--include-unsafe-tests` escape hatch for integration testing, we need some other mechanism to solve those use cases for developers.

Reply:
I struggle a bit with the concept of an "unsafe test". I can understand unsafe provisioning then running the test in that environment. Provisioning is inherently mutating important state. I worry that as soon as you frame a _test_ as unsafe, it's fundamentally trying to modify things that could (a) impact other tests or (b) impact users.